client/dashboard/app/queries/site-backup-restore.ts

```diff
@@ -0,0 +1,47 @@
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
+import {
+	fetchSiteBackupRestoreProgress,
+	initiateSiteBackupRestore,
+	type RestoreProgress,
+	type RestoreConfig,
+} from '../../data/site-backup-restore';
+import { queryClient } from '../query-client';
+
+/**
+ * Fetch the restore progress for a site.
+ * @param siteId - The ID of the site to fetch restore progress for.
+ * @param restoreId - The ID of the restore to fetch progress for.
+ * @returns A promise that resolves to the restore progress.
+ */
+export const siteBackupRestoreProgressQuery = ( siteId: number, restoreId: number ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'backup', 'restore', restoreId, 'progress' ],
+		queryFn: () => fetchSiteBackupRestoreProgress( siteId, restoreId ),
+		enabled: !! restoreId,
+		refetchInterval: ( query: { state: { data?: RestoreProgress } } ) => {
```

Comment:
I'm curious about whether the type is required or not. I guess `refetchInterval` knows the type automatically 🤔